src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -80,29 +82,47 @@ private static (IReadOnlyDictionary<FieldDeclaration, InputModelProperty> Fields
                 var fieldModifiers = inputModelProperty.IsReadOnly || inputModelProperty.Type is InputDictionaryType or InputListType
                     ? Public | ReadOnly
                     : Public;
+                var fieldType = GetDefaultPropertyType(inputModelProperty, typeFactory);
 
-                var field = new FieldDeclaration($"{inputModelProperty.Description}", fieldModifiers, typeFactory.CreateType(inputModelProperty.Type), inputModelProperty.Name.FirstCharToUpperCase(), writeAsProperty: true);
+                var field = new FieldDeclaration($"{inputModelProperty.Description}", fieldModifiers, fieldType, inputModelProperty.Name.FirstCharToUpperCase(), writeAsProperty: true);
                 fieldsToInputs[field] = inputModelProperty;
                 if (inputModelProperty.IsRequired)
                 {
-                    var parameter = Parameter.FromModelProperty(inputModelProperty, typeFactory);
+                    var parameter = Parameter.FromModelProperty(inputModelProperty, fieldType);
                     parametersToFields[parameter.Name] = field;
                     serializatoinParameters.Add(parameter);
+                    if (!inputModelProperty.IsReadOnly)
+                    {
+                        publicParameters.Add(parameter);
+                    }
                 }
             }
 
-            return (fieldsToInputs, serializatoinParameters, parametersToFields);
+            return (fieldsToInputs, publicParameters, serializatoinParameters, parametersToFields);
         }
 
-        private static (ConstructorSignature PublicConstructor, ConstructorSignature SerializationConstructor) BuildConstructors(string name, IReadOnlyList<Parameter> serializationParameters)
+        private static CSharpType GetDefaultPropertyType(in InputModelProperty property, TypeFactory typeFactory)
+        {
+            var valueType = typeFactory.CreateType(property.Type);
+
+            if (//!_usage.HasFlag(SchemaTypeUsage.Input) ||
```

Comment:
I comment out the codes, but I think later we will handle output model case as well.